appshell/appshell_extensions_win.cpp

```diff
@@ -939,7 +1055,20 @@ int32 WriteFile(ExtensionString filename, std::string contents, ExtensionString
     if (INVALID_HANDLE_VALUE == hFile)
         return ConvertWinErrorCode(GetLastError(), false); 
 
-    // TODO (issue 67) -  Should write to temp file and handle encoding
+    // TODO (issue 67) -  Should write to temp file
+    if (encoding != L"UTF-8") {
+        std::wstring content = StringToWString(contents);
+        std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> conv;
+        CharSetMap::iterator iter = charSetMap.find(conv.to_bytes(encoding));
+        if (iter != charSetMap.end()) {
+            WideToCharSet(content, iter->second, contents);
+        }
+        else {
+            return ERR_UNSUPPORTED_ENCODING;
```

Comment:
Handle left unclosed. If you return from here, you are not calling `CloseHandle()`. you need to do use `RAII pattern`.